scipy/optimize/cython_optimize/examples/zeros_struct_examples.pyx

```diff
@@ -0,0 +1,89 @@
+"""
+Use a structure to hold callback args so there's no Python and functions can
+free the global interpreter lock.
+"""
+
+from __future__ import division, print_function, absolute_import
+from math import exp, sin
+
+from .. cimport zeros_struct
+
+# test parameter structure
+ctypedef struct test_params:
+    double voltage
+    double light_current
+    double dark_current
+    double series_resistance
+    double shunt_resistance
+    double thermal_voltage
+
+NUM_OF_IRRAD = 10
+IL = [sin(il) + 6.0 for il in range(NUM_OF_IRRAD)]
+
+
+# governing equations
+cdef double f_solarcell(double i, void *args):
+    cdef test_params *myargs
+    cdef double v, il, io, rs, rsh, vt
+    # unpack structure
+    myargs = <test_params *> args
+    v = myargs.voltage
+    il = myargs.light_current
+    io = myargs.dark_current
+    rs = myargs.series_resistance
+    rsh = myargs.shunt_resistance
+    vt = myargs.thermal_voltage
+    vd = v + i * rs
+    return il - io * (exp(vd / vt) - 1.0) - vd / rsh - i
+
+
+cdef double fprime(double i, void *args):
+    cdef test_params *myargs
+    cdef double v, il, io, rs, rsh, vt
+    # unpack structure
+    myargs = <test_params *> args
+    v = myargs.voltage
+    il = myargs.light_current
+    io = myargs.dark_current
+    rs = myargs.series_resistance
+    rsh = myargs.shunt_resistance
+    vt = myargs.thermal_voltage
+    return -io * exp((v + i * rs) / vt) * rs / vt - rs / rsh - 1
+
+
+#solver
+cdef double solarcell_newton(tuple args):
```

Comment:
why not use dictionary here and in [`solarcell_bisect()`](#diff-c2d8e3e9fc487db281e9aee4829635a7R74) instead of tuple?